src/fetch-map/layer-map.ts

```diff
@@ -13,24 +13,19 @@ import {
 import {format as d3Format} from 'd3-format';
 import moment from 'moment-timezone';
 
-import {Accessor, Layer, _ConstructorOf as ConstructorOf} from '@deck.gl/core';
-import {
-  GridLayer,
-  HeatmapLayer,
-  HexagonLayer,
-} from '@deck.gl/aggregation-layers';
-import {GeoJsonLayer} from '@deck.gl/layers';
-import {H3HexagonLayer} from '@deck.gl/geo-layers';
-import {
-  ClusterTileLayer,
-  H3TileLayer,
-  QuadbinTileLayer,
-  RasterTileLayer,
-  VectorTileLayer,
-  HeatmapTileLayer,
-} from '@deck.gl/carto';
-
-import {assert} from '../utils.js';
+import type {Accessor, Layer, _ConstructorOf as ConstructorOf} from '@deck.gl/core';
+import type { ClusterTileLayer, H3TileLayer, QuadbinTileLayer, RasterTileLayer, VectorTileLayer, HeatmapTileLayer } from '@deck.gl/carto';
+
+export type LayerProvider = {
+  clusterTile: ConstructorOf<ClusterTileLayer>,
+  h3: ConstructorOf<H3TileLayer>,
+  heatmapTile: ConstructorOf<HeatmapTileLayer>,
+  mvt: ConstructorOf<VectorTileLayer>,
+  quadbin: ConstructorOf<QuadbinTileLayer>,
+  raster: ConstructorOf<RasterTileLayer>,
+  tileset: ConstructorOf<VectorTileLayer>
+}
```

Comment:
Do the keys of this object come from somewhere particular? I might be inclined to make these particular types less strict...

```typescript
// (A)
export type LayerProvider = Record<'clusterTile' | 'h3' | ..., ConstructorOf<Layer>>;

// (B)
export type LayerProvider = Record<'clusterTile' | 'h3' | ..., unknown>;
```

... otherwise TS is potentially doing deep compatibility checks here that will just make version upgrades, layer overrides, and dual package hazard more risky, without much type safety benefit.

Reply:
The keys are based on what the API gives us, via the `keplerMapConfig` (with CARTO extensions). I've gone ahead and loosened the value to `Layer`. This gives us the option to swap out an arbitrary `Layer` and just have it receive the props, thus not forcing the consumer to subclass a CARTO layer if they don't want to (although I expect most will)

Reply:
The extra benefit is that we don't need to have `@deck.gl/carto` as a peerDependency and we can just use `@deck.gl/core` which is more lightweight